netlogo-gui/src/main/lab/Worker.scala

```diff
@@ -119,14 +149,14 @@ class Worker(val protocol: LabProtocol)
             ws.world.mainRNG.setSeed(value.asInstanceOf[java.lang.Double].longValue)
         }
```

Comment:
Another option instead of doing all the checking above is simply to see what value `d` takes when it's finished being built at this point (this would probably be my preferred approach). It means you have to do little to nothing but do a check on each of the min/max coordinates here at the end. It would be a good deal shorter and with less crazy looking through lists.